### PR TITLE
UAR-104: Fix for nav check, hasPresenter reinstated, hasUpdatePresenter new code

### PR DESCRIPTION
**Pull request by corybudd-ch**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-104

### Change description
hasPresenter and hasUpdatePresenter using shared code doesn't work. EntityNameKey doesn't exist in update journey.

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/middleware/navigation/check.condition.ts [hunk 1]
```diff
@@ -68,6 +68,10 @@ export const checkOverseasEntityNumberEntered = (appData: ApplicationData): bool
   return checkHasAppData(appData) && (appData[OeNumberKey] || "").length !== 0;
 };
 
+export const checkUpdatePresenterEntered = (appData: ApplicationData): boolean => {
+  return checkOverseasEntityNumberEntered(appData) && Object.keys(appData[PresenterKey] || {}).length !== 0;
```
Comment (AndreKramer7654) [inline src/middleware/navigation/check.condition.ts:72]: Should this not check  checkHasAppData(appData) && ...?

Comment (corybudd-ch) [reply src/middleware/navigation/check.condition.ts:72]: The `checkOverseasEntityNumberEntered` does that as its first condition.

```
export const checkOverseasEntityNumberEntered = (appData: ApplicationData): boolean => {
  return checkHasAppData(appData) && (appData[OeNumberKey] || "").length !== 0;
};
```

Comment (AndreKramer7654) [reply src/middleware/navigation/check.condition.ts:72]: But why check checkOverseasEntityNumberEntered?

File: src/middleware/navigation/has.presenter.middleware.ts [hunk 2]
```diff
@@ -1,8 +1,18 @@
 import { Request, Response, NextFunction } from 'express';
 
-import { checkIfPresenterDataExists } from './checks/has.presenter';
+import { logger } from '../../utils/logger';
 import { SOLD_LAND_FILTER_URL } from '../../config';
+import { getApplicationData } from "../../utils/application.data";
+import { checkPresenterDetailsEntered, NavigationErrorMessage } from './check.condition';
 
 export const hasPresenter = (req: Request, res: Response, next: NextFunction): void => {
-  checkIfPresenterDataExists(req, res, next, SOLD_LAND_FILTER_URL);
```
Comment (AndreKramer7654) [inline src/middleware/navigation/has.presenter.middleware.ts:7]: Are we removing checkIfPresenterDataExists?

Comment (AndreKramer7654) [reply src/middleware/navigation/has.presenter.middleware.ts:7]: from another file?

File: src/middleware/navigation/check.condition.ts (hunk 1 above)
Comment (jkayyala) [reply src/middleware/navigation/check.condition.ts:72]: @AndreKramer7654 , this is checking OeNumberKey, as that is what is stored in the session after a successful search from overseas-entity-query page

**Review by AndreKramer7654 — APPROVED** [review]

**Review by jkayyala — APPROVED** [review]

**Review by olaych — APPROVED** [review]

**Review by olaych — COMMENTED** [review]
Approved but who is making changes doesn't look right

**Pull request merged**